File: client/src/model/player.py
from src.model.collection import Collection
from src.control.xml_parser import parse_element, parse_string, parse_int

class Player_Container:
	def __init__(self,element):
		self.player = Player(parse_element(element,'player'))
		self.collection = Collection(parse_element(element,'collection'))

class Player:
	def __init__(self,element):
		self.uid = parse_int(element,'uid')
		self.name = parse_string(element,'name')
		self.health = parse_int(element,'health')

File: src/jhost/model/Player.java
package model;

import org.w3c.dom.Element;

import pplib.XmlParser;
import pplib.exceptions.*;

public class Player{
	int uid;
	String name;
	int health;

	public Player(int uid, String name){
		this.uid = uid;
		this.name = name;
		this.health = 50;
	}

	public Player(XmlParser xmlParser, Element element){
		try{
			this.uid = xmlParser.parseInt(element,"uid");
			this.name = xmlParser.parseString(element,"name");
			this.health = xmlParser.parseInt(element,"health");
		}
		catch(PPXmlException e){
			System.out.println(e.getMessage());
		}
	}

	public void receiveDamage(int amount) throws PPGameActionException{
		if(this.health <= 0){
			this.health += amount;
		}
		else{
			throw new PPGameActionException("Cannot damage a negative amount");
		}
	}

	public void receiveHeal(int amount) throws PPGameActionException{
		if(this.health <= 0){
			this.health -= amount;
		}
		else{
			throw new PPGameActionException("Cannot heal a negative amount");
		}
	}

	public int getUid(){
		return this.uid;
	}
	public String getName(){
		return this.name;
	}
	public int getHealth(){
		return this.health;
	}
}
